productivity-suite/app/fragments/news/src/routes/[...news].tsx

```diff
@@ -3,23 +3,47 @@ import {
   createResource,
   createSignal,
   For,
+  onMount,
   Show,
   Suspense,
+  createEffect,
 } from "solid-js";
 import Story from "~/components/story";
 import fetchAPI from "~/lib/api";
 import { IStory } from "~/types";
+import { getBus, getFragmentHost } from "piercing-library";
 
 export default function News() {
+  let ref: HTMLDivElement;
+
   const [page, setPage] = createSignal<number>(1);
 
+  onMount(() => {
+    getBus(ref).listen<{ page: number }>("news-page", ({ page }) => {
+      if (isFinite(page) && page > 0) {
+        setPage(page);
+      }
+    });
+  });
+
+  onMount(() => {
+    getFragmentHost(ref)?.onCleanup(() =>
```

Comment:
In this case the fragment host should always be present, so it wouldn't be more correct to have:
```suggestion
    getFragmentHost(ref)!.onCleanup(() =>
```
?

Reply:
Yes, I guess this is similar to our discussion today, where we expect fragments to always be run inside a context that has a message bus and a host...